### PR TITLE
Add director client for dashboard URL request

### PR DESCRIPTION
**Pull request by jasiu001**
Related to #600 
additionally, integration with golang-migration tool has been added

### CODE REVIEW

File: components/kyma-environment-broker/internal/broker/broker.go [hunk 1]
```diff
@@ -305,6 +301,42 @@ func (b *KymaEnvBroker) LastOperation(ctx context.Context, instanceID string, de
 	}, nil
 }
 
+func (b *KymaEnvBroker) handleDashboardURL(instance *internal.Instance) domain.LastOperationState {
+	b.Dumper.Dump("Get dashboard url for instance ID: ", instance.InstanceID)
+
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/broker/broker.go]: maybe we could check if there is a dashboard url in the instance, if not - then ask director for it

File: components/kyma-environment-broker/internal/broker/broker.go [hunk 2]
```diff
@@ -262,31 +272,25 @@ func (b *KymaEnvBroker) LastOperation(ctx context.Context, instanceID string, de
...
-		if err != nil {
-			b.Dumper.Dump("Got error: ", err)
-		}
-		inst.DashboardURL = fixedDummyURL
-		err = b.InstancesStorage.Update(*inst)
-		if err != nil {
-			b.Dumper.Dump("Got error: ", err)
-		}
-
+		lastOpStatus = b.handleDashboardURL(instance)
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/broker/broker.go]: when the provisioner returns Success but  there is no dashboard url we get at the endpoint:
status "in progress" but description "success"

File: components/schema-migrator/validate.sh [hunk 3]
```diff
@@ -44,32 +45,48 @@ docker run -d --name ${POSTGRES_CONTAINER} \
...
+            -e NON_INTERACTIVE="true" \
+        ${IMG_NAME}
+
+    echo -e "${GREEN}Show schema_migrations table after DOWN migrations${NC}"
+    docker exec ${POSTGRES_CONTAINER} psql -U usr compass -c "select * from schema_migrations"
+}
+
+for i in "${MIGRATION_PATHS[@]}"
+do
+   echo -e "${GREEN}Migrations for \"${i}\" path${NC}"
```
Comment (aszecowka) [inline components/schema-migrator/validate.sh]: you wrote to the same database and to the same schema_migrations table - keep it separate.

File: components/kyma-environment-broker/internal/director/client.go [hunk 4]
```diff
@@ -1,33 +1,161 @@
...
+	token, err := dc.oauthClient.GetAuthorizationToken()
+	if err != nil {
+		return errors.Wrap(err, "Error while obtaining token")
+	}
+	dc.token = token
+
 	return nil
 }
+
+func (dc Client) getURLFromRuntime(response *graphql.RuntimeExt) (string, error) {
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/director/client.go]: use pointer receiver

File: components/schema-migrator/migrations/kyma-environment-broker/202001231030_add_datetimes_to_instances.up.sql [hunk 5]
```diff
@@ -0,0 +1,8 @@
+ALTER TABLE instances
+    ADD COLUMN created_at TIMESTAMPTZ NOT NULL DEFAULT NOW();
+
+ALTER TABLE instances
+    ADD COLUMN updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW();
+
+ALTER TABLE instances
```
[the remark this answers is not in the document]
Comment (jasiu001) [reply components/schema-migrator/migrations/kyma-environment-broker/202001231030_add_datetimes_to_instances.up.sql]: instance model has field `DelatedAt` as [time.Time](https://github.com/kyma-incubator/compass/blob/f9bd6c3025586b54524375adc681cc3781ce8531/components/kyma-environment-broker/internal/model.go#L17) library which we use github.com/gocraft/dbr cannot handle empty datetime from postgres

File: components/kyma-environment-broker/internal/director/client.go (hunk 4 above)
Comment (jasiu001) [reply components/kyma-environment-broker/internal/director/client.go]: I changed but IMO it is not necessary, inside method any of client field is not changed.

File: components/kyma-environment-broker/internal/director/oauth/client.go [hunk 6]
```diff
@@ -9,14 +10,11 @@ import (
 	"strings"
 	"time"
 
-	"golang.org/x/net/context"
-	v1 "k8s.io/api/core/v1"
-
 	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
```
[the remark this answers is not in the document]
Comment (jasiu001) [reply components/kyma-environment-broker/internal/director/oauth/client.go]: OAuth client was introduced in this [PR](https://github.com/kyma-incubator/compass/pull/635), I used it in my code.

File: components/schema-migrator/validate.sh (hunk 3 above)
Comment (jasiu001) [reply components/schema-migrator/validate.sh]: I used `Initialization scripts` ([link](https://hub.docker.com/_/postgres)) to create databases

File: chart/compass/templates/migrator-job.yaml [hunk 7]
```diff
@@ -76,7 +78,92 @@ spec:
...
+        "helm.sh/hook-weight": "1"
+        "helm.sh/hook-delete-policy": before-hook-creation
+spec:
+    template:
+        metadata:
+            labels:
+                app: {{ .Chart.Name }}
+                release: {{ .Release.Name }}
+        spec:
+            restartPolicy: Never
```
Comment (aszecowka) [inline chart/compass/templates/migrator-job.yaml]: set restartPolicy to OnFailure for both jobs

File: components/schema-migrator/postgres/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,3 @@
+FROM postgres:11
```
Comment (aszecowka) [inline components/schema-migrator/postgres/Dockerfile]: why do you create another Docker image? you can mount the script to docker-entrypoint-initdb.d when running 
 ordinary postgres image.

**Pull request merged**